server/eventlog.go

```diff
@@ -62,6 +74,54 @@ func contains(set [][]byte, value []byte) bool {
 	return false
 }
 
+func getVerifiedCosState(celCosBytes []byte, pcrs *tpmpb.PCRs) (*pb.AttestedCosState, error) {
+	coscel, err := cel.DecodeToCEL(bytes.NewBuffer(celCosBytes))
+	if err != nil {
+		return nil, err
+	}
+
+	// Validate the COS event log first.
+	if err := coscel.Replay(pcrs); err != nil {
+		return nil, err
+	}
+
+	cosState := &pb.AttestedCosState{}
+	cosState.Container = &pb.ContainerState{}
+
+	for _, record := range coscel.Records {
+		// ignore non COS CEL events
+		if !record.Content.IsCosTlv() {
+			continue
+		}
+
+		cosTlv, err := record.Content.ParseToCosTlv()
+		if err != nil {
+			return nil, err
+		}
+
+		// verify digests for the cos cel content
+		if err := cel.VerifyDigest(cosTlv, record.Digests); err != nil {
+			return nil, err
+		}
+
+		switch cosTlv.EventType {
+		case cel.ImageRefType:
+			cosState.Container.ImageReference = string(cosTlv.EventContent)
+
+		case cel.ImageDigestType:
+			cosState.Container.ImageDigest = cosTlv.EventContent
+
+		case cel.RestartPolicyType:
+			restartPolicy, ok := pb.RestartPolicy_value[string(cosTlv.EventContent)]
```

Comment:
So are the measured values in the CEL log going to look like: `"OnFailure"` (the style from [Pod Lifecycle](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/)) or `"ON_FAILURE"` (the style from [our protobuf](https://github.com/google/go-tpm-tools/blob/bc3fe1ecfe70da837b051fc8d81dfaa22bfa5170/proto/attest.proto#L114-L122))

Reply:
It will use protobuf ON_FAILURE, when the launcher is writing the log, it needs to use the protobuf enum string.

Reply:
Ok, I have an idea about how to make this more ergonomic, but that can be in a followup PR